libraries/AP_NMEA_Output/AP_NMEA_Output.h

```diff
@@ -50,6 +50,8 @@ class AP_NMEA_Output {
     AP_HAL::UARTDriver* _uart[NMEA_MAX_OUTPUTS];
 
     uint32_t _last_run_ms;
+    char _gga[74];
+    char _rmc[108];
```

Comment:
Problem here is that we permanently lose these bytes, and RAM is in short supply.  What was your reasoning behind this change?

What was your reasoning behind the change?  All the overhead of dynamic allocation?  Perhaps an argument could be for allocating once as soon as the module is used....

Reply:
@peterbarker 
The idea is to fail as soon as possible. So if there is not enough RAM available it just can't be loaded. But if loaded all features should be set up and ready for action even if currently not used.  This is what I prefer at least for systems like FC. 
In fact it also safes RAM because dynamic allocated blocks do have at least one additional char* pointer for keeping them in the list of used  blocks. Not allocating two blocks does save at least 8 bytes.

Also there is no public constructor, but probe() is a factory function creating the object with  `new`  anyway.

Reply:
@peterbarker On the off chance you missed it [here's the original issue that requests this change](https://github.com/ArduPilot/ardupilot/issues/12514).  Both you and Tridge are involved!

Reply:
I had a closer look to the generated binary and both size and elf headers are not affected by the size of the buffers, so what I can tell this change has no impact on the used RAM size if the feature is not used.